stl/inc/experimental/random

```diff
@@ -0,0 +1,55 @@
+// random experimental header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _EXPERIMENTAL_RANDOM_
+#define _EXPERIMENTAL_RANDOM_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#include <random>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+namespace experimental {
+    inline namespace fundamentals_v3 {
+
+        inline default_random_engine& _Engine() {
+            thread_local default_random_engine _Eng{random_device{}()};
+            return _Eng;
+        }
+
+        template <class _IntType>
+        _IntType randint(_IntType _Min, _IntType _Max) {
+            static_assert(_Is_any_of_v<remove_cv_t<_IntType>, short, int, long, long long, unsigned short, unsigned int,
+                              unsigned long, unsigned long long>,
+                "template argument does not meet the requirements for IntType.");
+            _STL_ASSERT(_Min <= _Max, "_Min > _Max");
+            return uniform_int_distribution<_IntType>{_Min, _Max}(_Engine());
+        }
+
+        inline void reseed() {
+            _Engine().seed(random_device{}());

```

Comment:
This should be default_seed

Reply:
Then that wouldn't satisfy "The first form sets g to an unpredictable state". Default seed os predictable.

Reply:
Hm I took this from the actual random library. It might be that they differ but maybe a maintainer can shed some light here.

Looking at the specification you linked to it seems that you need to reset the uniform_int_Distribution. At least that is what the note says.

Reply:
They do differ the standard says, `seed()` is equivalent to `E()` and that "Creates an engine with the same initial state as all other default-constructed engines of type E"

https://eel.is/c++draft/rand.req.eng#tab:rand.req.eng

Reply:
Yeah, I concure with Thatcher. This leaves only the reset of the distribution

Reply:
Maybe I'm missing something but Isn't that taken care of by the fact that a new instance of `uniform_int_distribution` is created whenever `randint` is called?

Reply:
I think you are safe in the reseed case. However, in the usual case the result might differ, depending on whether the implementation of the distribution is statefull.

Imagine the simple case of int. The random_engine gives 64 bits of entropy. You only need 32 of those. So rather than throwing them away you can reuse them for the next call.

However, you are lucky as uniform_int_distribution is stateless

Reply:
> You only need 32 of those. So rather than throwing them away you can reuse them for the next call.

`uniform_int_distribution` can't do that because different calls to operator() may be passed different engines; it would be improper if the result depended on any engine other than that passed to operator().